extern/boostd-data/yugabyte/service.go

```diff
@@ -442,54 +450,63 @@ func (s *Store) addMultihashesToPieces(ctx context.Context, pieceCid cid.Cid, re
 	ctx, span := tracing.Tracer.Start(ctx, "store.add_index.payloadpiece")
 	defer span.End()
 
-	var count float64
-	return s.execParallel(ctx, recs, s.settings.PayloadPiecesParallelism, func(rec model.Record) error {
-		multihashBytes := rec.Cid.Hash()
-		q := `INSERT INTO PayloadToPieces (PayloadMultihash, PieceCid) VALUES (?, ?)`
-		err := s.session.Query(q, trimMultihash(multihashBytes), pieceCid.Bytes()).Exec()
-		if err != nil {
-			return fmt.Errorf("inserting into PayloadToPieces: %w", err)
+	insertPieceOffsetsQry := `INSERT INTO PayloadToPieces (PayloadMultihash, PieceCid) VALUES (?, ?)`
+	pieceCidBytes := pieceCid.Bytes()
+
+	var batch *gocql.Batch
+	for allIdx, rec := range recs {
+		if batch == nil {
+			batch = s.session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
+			batch.Entries = make([]gocql.BatchEntry, 0, s.settings.InsertBatchSize)
 		}
 
-		count++
-		progress(count / float64(len(recs)))
-		return nil
-	})
+		batch.Entries = append(batch.Entries, gocql.BatchEntry{
+			Stmt:       insertPieceOffsetsQry,
+			Args:       []interface{}{trimMultihash(rec.Cid.Hash()), pieceCidBytes},
+			Idempotent: true,
+		})
+
+		if allIdx == len(recs)-1 || len(batch.Entries) == s.settings.InsertBatchSize {
+			err := s.session.ExecuteBatch(batch)
+			if err != nil {
+				return fmt.Errorf("inserting into PayloadToPieces: %w", err)
+			}
+			batch = nil
+
+			progress(float64(allIdx+1) / float64(len(recs)))
+		}
+	}
+	return nil
 }
 
 func (s *Store) addPieceInfos(ctx context.Context, pieceCid cid.Cid, recs []model.Record, progress func(addProgress float64)) error {
 	ctx, span := tracing.Tracer.Start(ctx, "store.add_index.pieceinfo")
 	defer span.End()
 
-	batchEntries := make([]gocql.BatchEntry, 0, len(recs))
 	insertPieceOffsetsQry := `INSERT INTO PieceBlockOffsetSize (PieceCid, PayloadMultihash, BlockOffset, BlockSize) VALUES (?, ?, ?, ?)`
-	for _, rec := range recs {
-		batchEntries = append(batchEntries, gocql.BatchEntry{
-			Stmt:       insertPieceOffsetsQry,
-			Args:       []interface{}{pieceCid.Bytes(), rec.Cid.Hash(), rec.Offset, rec.Size},
-			Idempotent: true,
-		})
-	}
+	pieceCidBytes := pieceCid.Bytes()
 
-	// The Cassandra driver has a 50k limit on batch statements. Keeping
-	// batch size small makes sure we're under the limit.
-	const batchSize = 5000
 	var batch *gocql.Batch
-	for allIdx, entry := range batchEntries {
+	for allIdx, rec := range recs {
 		if batch == nil {
 			batch = s.session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
+			batch.Entries = make([]gocql.BatchEntry, 0, s.settings.InsertBatchSize)
```

Comment:
ditto re slice size.

Would be nice to reduce duplicate code in working with batches.